src/Transform/ONNX/ConstProp.cpp

```diff
@@ -93,12 +94,10 @@ ElementsAttr getConstValueElements(Value constValue) {
   return constOp.getValueAttr().cast<ElementsAttr>();
 }
 
-// Creates ONNXConstantOp with the location and result type from replacingValue.
-ONNXConstantOp createReplacingConstantOp(
+// Creates ONNXConstantOp with the location from replacingValue.
+Value createReplacingConstantOp(
     PatternRewriter &rewriter, Value replacingValue, ElementsAttr elements) {
-  return rewriter.create<ONNXConstantOp>(replacingValue.getLoc(),
-      replacingValue.getType(), Attribute(), elements, FloatAttr(), ArrayAttr(),
-      IntegerAttr(), ArrayAttr(), StringAttr(), ArrayAttr());
+  return OnnxBuilder(rewriter, replacingValue.getLoc()).constant(elements);
 }
```

Comment:
Note that this change means that the Constant op gets the type from elements attribute instead of replacingValue. In the past we had a couple of cases where these types disagreed (the Constant had unranked type or a dynamic dim) but I think we've cleared up all those cases. We need the Constant op type to agree with its attribute to make constant folding work going forward.